### ISSUE TITLE
Why the loss is weighted in compile() ?

### ISSUE THREAD

**Issue by jedi00** [fchollet/keras#519]
@wxs , @fchollet : I'm trying to transplant CTC objective into Keras. When reading the codes, I noticed in `compile()` function of `Sequential` class,  there are lines as
```
        weighted_loss = weighted_objective(objectives.get(loss))   # return a function
        ...
        self.weights = T.ones_like(self.y_train)
        train_loss = weighted_loss(self.y, self.y_train, self.weights)
        test_loss = weighted_loss(self.y, self.y_test, self.weights)
```
Since the `self.weights' are already all ones, why there's need for weighting anyway?

And, in `weighted_objective` function, shapes of `masked_y_true` and `masked_y_pred` are different from `y_true` and `y_pred`, does this mean the objective function should expect the shapes of masked variables instead of unmasked ones?

**Comment by jedi00** [comment]
Anybody?

**Comment by fchollet** [comment]
To support the `sample_weight` and `class_weight` features. The exact process can be found in the code.
